clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
Good point.

1. Subscribe(empty) should still trigger a rebalance by sending a join-group request with empty subscription, so that it will then be assigned with no partition, but still live in the group;
2. Unsubscribe() should, in the future, trigger a rebalance by sending a leave-group request and reset its generation.
3. Pause() does not trigger a rebalance, but just stopping the fetching of some assigned partitions.

We can leave this in the future ticket.